Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -5745,11 +5753,24 @@ bool CSSPropertyParser::consumeBorderImage(CSSPropertyID property, bool importan
             return true;
         case CSSPropertyBorderImage:
         case CSSPropertyWebkitBorderImage:
-            addPropertyWithImplicitDefault(CSSPropertyBorderImageSource, property, WTFMove(source), valuePool.createImplicitInitialValue(), important);
-            addPropertyWithImplicitDefault(CSSPropertyBorderImageSlice, property, WTFMove(slice), valuePool.createImplicitInitialValue(), important);
-            addPropertyWithImplicitDefault(CSSPropertyBorderImageWidth, property, WTFMove(width), valuePool.createImplicitInitialValue(), important);
-            addPropertyWithImplicitDefault(CSSPropertyBorderImageOutset, property, WTFMove(outset), valuePool.createImplicitInitialValue(), important);
-            addPropertyWithImplicitDefault(CSSPropertyBorderImageRepeat, property, WTFMove(repeat), valuePool.createImplicitInitialValue(), important);
+            addPropertyWithImplicitDefault(CSSPropertyBorderImageSource, property, WTFMove(source), valuePool.createIdentifierValue(CSSValueNone), important);
+            if (slice)
+                addProperty(CSSPropertyBorderImageSlice, property, slice.releaseNonNull(), important);
+            else {
+                auto value = CSSBorderImageSliceValue::create(createQuad(100, CSSUnitType::CSS_PERCENTAGE), property == CSSPropertyWebkitBorderImage);
```

Comment:
Well spotted, `ApplyPropertyBorderImageModifier::applyInitialValue` doesn't set `fill` so there was this bug:

```js
var s = document.body.style;
var cs = getComputedStyle(document.body);
s.webkitBorderImage = "url()";
cs.borderImageSlice; // "100%"
s.webkitBorderImage = cs.webkitBorderImage; // "url(\"\") 100% / 1 / 0 stretch"
cs.borderImageSlice; // "100% fill"
```

Now it's consistent:

```js
s.webkitBorderImage = "url()";
cs.borderImageSlice; // "100% fill"
s.webkitBorderImage = cs.webkitBorderImage; // "url(\"\") 100% fill / 1 / 0 stretch"
cs.borderImageSlice; // "100% fill"
```

Reply:
Further thinking about this, the serialization of the computed value will still need to be fixed anyways, and this introduces a regression:

Previously, `-webkit-border-image` had round-tripping for slice:

```js
var s = document.body.style;
s.borderImage = "url()";
[s.borderImageSlice, // "initial"
 s.webkitBorderImage += "", // "url(\"\")"
 s.borderImageSlice]; // "initial"
```
Now it serializes if the slice is implicit, even if it doesn't have fill, and when reparsing it adds fill:

```js
var s = document.body.style;
s.borderImage = "url()";
[s.borderImageSlice, // "100%"
 s.webkitBorderImage += "", // "url(\"\")"
 s.borderImageSlice]; // "100% fill"
```

So if `-webkit-border-image` when resets an omitted `border-image-slice` it adds `fill`, then the serialization code should return empty string when there is a mismatch.

So maybe just keep the old behavior of not setting `fill` when omitted.

Reply:
OK, done.

Reply:
Filed https://bugs.webkit.org/show_bug.cgi?id=248113 for fixing the serialization of the computed `-webkit-border-image`